pkg/component/kubernetes/apiserverexposure/sni.go

```diff
@@ -89,51 +108,90 @@ type sni struct {
 	valuesFunc func() *SNIValues
 }
 
-type envoyFilterTemplateValues struct {
+type envoyFilterAPIServerProxyTemplateValues struct {
 	*APIServerProxy
 	IngressGatewayLabels        map[string]string
 	Name                        string
 	Namespace                   string
 	Host                        string
+	MutualTLSHost               string
 	Port                        int
 	APIServerClusterIPPrefixLen int
 }
 
+type envoyFilterIstioTLSTerminationTemplateValues struct {
+	AuthenticationDynamicMetadataKey string
+	Hosts                            []string
+	IngressGatewayLabels             map[string]string
+	Name                             string
+	Namespace                        string
+	MutualTLSHost                    string
+	Port                             int
+	RouteConfigurationName           string
+}
+
 func (s *sni) Deploy(ctx context.Context) error {
 	var (
 		values = s.valuesFunc()
 
-		destinationRule = s.emptyDestinationRule()
-		gateway         = s.emptyGateway()
-		virtualService  = s.emptyVirtualService()
+		destinationRule     = s.emptyDestinationRule()
+		mTLSDestinationRule = s.emptyMTLSDestinationRule()
+		gateway             = s.emptyGateway()
+		virtualService      = s.emptyVirtualService()
 
-		hostName        = fmt.Sprintf("%s.%s.svc.%s", s.name, s.namespace, gardencorev1beta1.DefaultDomain)
-		envoyFilterSpec bytes.Buffer
+		hostName                       = fmt.Sprintf("%s.%s.svc.%s", s.name, s.namespace, gardencorev1beta1.DefaultDomain)
+		mTLSHostName                   = fmt.Sprintf("%s.%s.svc.%s", s.name+MutualTLSServiceNameSuffix, s.namespace, gardencorev1beta1.DefaultDomain)
+		envoyFilterAPIServerProxy      bytes.Buffer
+		envoyFilterIstioTLSTermination bytes.Buffer
 	)
 
+	registry := managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
+
 	if values.APIServerProxy != nil {
-		envoyFilter := s.emptyEnvoyFilter()
+		envoyFilter := s.emptyEnvoyFilterAPIServerProxy()
 		apiServerClusterIPPrefixLen, err := netutils.GetBitLen(values.APIServerProxy.APIServerClusterIP)
 		if err != nil {
 			return err
 		}
 
-		if err := envoyFilterSpecTemplate.Execute(&envoyFilterSpec, envoyFilterTemplateValues{
+		if err := envoyFilterAPIServerProxyTemplate.Execute(&envoyFilterAPIServerProxy, envoyFilterAPIServerProxyTemplateValues{
 			APIServerProxy:              values.APIServerProxy,
 			IngressGatewayLabels:        values.IstioIngressGateway.Labels,
 			Name:                        envoyFilter.Name,
 			Namespace:                   envoyFilter.Namespace,
 			Host:                        hostName,
 			Port:                        kubeapiserverconstants.Port,
+			MutualTLSHost:               mTLSHostName,
```

Comment:
Why was this field added? At least in this commit, I do not see any usage in the template.

Reply:
This is a left-over. We initially added the used the existing template, then split them and I forgot to remove this field.